elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,375 @@
+# Official documentation on making custom PL
+
+
+def render_html_colour(score):
+    # used to render the correct colour depending on student score
+    if score == 0:
+        return 'badge-danger'
+    elif score == 1.0:
+        return 'badge-success'
+    else:
+        return 'badge-warning'
+
+def get_all_answer(submitted_blocks, block_indents, leading_code, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer_code = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer_code += ('    ' * indent) + answer + '\n'
+    answer_code = leading_code + '\n' + answer_code + trailing_code + '\n'
+    return answer_code
+
+def prepare(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    
+    pl.check_attribs(element,
+                 required_attribs=['answers-name'],
+                 optional_attribs=['shuffle-options',
+                                   'permutation-mode',
+                                   'check-indentation',
+                                   'header-left-column',
+                                   'header-right-column',
+                                   'external-grader',
+                                   'file-name',
+                                   'leading-code',
+                                   'trailing-code',
+                                   'dropzone-layout',
+                                   'max-incorrect',
+                                   'min-incorrect'])
+
+    answer_name = pl.get_string_attrib(element, 'answers-name')
+
+    mcq_options = []
+    correct_answers = []
+    correct_answers_indent = []
+    incorrect_answers = []
+
+    for html_tags in element: # iterate through the tags inside pl-order-blocks, should be <pl-answer> tags
+        if html_tags.tag == 'pl-answer':
+            # correct attribute is not strictly required, as the attribute is irrelevant for autograded questions
+            pl.check_attribs(html_tags, required_attribs=[], optional_attribs=['correct', 'ranking', 'indent'])
+
+            isCorrect = pl.get_boolean_attrib(html_tags, 'correct', True)  # default correctness to True
+            answerIndent = pl.get_string_attrib(html_tags, 'indent', '-1')  # get answer indent, and default to -1 (indent level ignored)
+            if isCorrect == True:
+                # add option to the correct answer array, along with the correct required indent
+                correct_answers.append(str.strip(html_tags.text))
+                correct_answers_indent.append(answerIndent)
+            else:
+                incorrect_answers.append(str.strip(html_tags.text))
+
+    if pl.get_boolean_attrib(element, 'external-grader', False) is False and len(correct_answers) == 0:
+        raise Exception('There are no correct answers specified for this question!')
+
+    minIncorrect = pl.get_integer_attrib(element, 'min-incorrect', 1)
```

Comment:
I also found incorrect behavior of the grading code when using the option `permutation-mode = html-order`. Since this option does not have partial credit, I think it is giving zero when not all the blocks are moved, however, sometimes, some of the blocks don't exist to be moved (and hence the answer is still correct)

Reply:
min and maxincorrect behaviour has been changed as you suggested

still working on the bug you mentioned